src/components/TodosList/TodosList.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect } from 'react';
+import './TodoList.scss';
+import classNames from 'classnames';
+import { useDispatch, useSelector } from 'react-redux';
+import { getTodos } from '../../api/todos';
+import {
+  loadTodosAction, loadUserAction, setInputValue, setSelectValue,
+} from '../../store/actions';
+import {
+  getInputValue, getSelectValue, getTodosSelector, getUserSelector,
+} from '../../store/selectors';
+import { getUser } from '../../api/user';
+import { Todo } from '../../react-app-env';
+
+export const TodosList: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const todos = useSelector(getTodosSelector);
+  const inputValue = useSelector(getInputValue);
+  const selectValue = useSelector(getSelectValue);
+  const user = useSelector(getUserSelector);
+
+  const handelSelectUser = async (userId: number) => {
+    const userFromServer = await getUser(userId);
+
+    dispatch(loadUserAction(userFromServer));
+  };
+
+  useEffect(() => {
+    const loadTodosFromServer = async () => {
+      const todosFromServer = await getTodos();
+
+      dispatch(loadTodosAction(todosFromServer));
+    };
+
+    loadTodosFromServer();
+  }, []);
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setSelectValue(event.target.value));
+  };
+
+  const filterTodos = (): Todo[] => {
+    let filteredTodos;
+
+    switch (selectValue) {
+      case 'active':
+        filteredTodos = todos.filter(todo => !todo.completed);
+        break;
+
+      case 'completed':
+        filteredTodos = todos.filter(todo => todo.completed);
+        break;
+
+      default:
+        filteredTodos = [...todos];
+    }
+
+    return filteredTodos.filter(todo => (
+      todo.title.toLowerCase().includes(inputValue.toLocaleLowerCase())
```

Comment:
create a separate variable for `inputValue.toLocaleLowerCase()`
no need to do this operation on every iteration